core/src/main/java/org/frankframework/pipes/ForPipe.java

```diff
@@ -0,0 +1,122 @@
+/*
+   Copyright 2024 WeAreFrank!
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package org.frankframework.pipes;
+
+import lombok.Getter;
+
+import org.frankframework.configuration.ConfigurationException;
+import org.frankframework.core.PipeForward;
+import org.frankframework.core.PipeLineSession;
+import org.frankframework.core.PipeRunException;
+import org.frankframework.core.PipeRunResult;
+import org.frankframework.doc.ElementType;
+import org.frankframework.doc.ElementType.ElementTypes;
+import org.frankframework.stream.Message;
+
+/**
+ * ForPipe is a wrapper to use another pipe a fixed number of times. This can be accomplished by something like:
+ *
+ * <pre>{@code
+ * 		<ForPipe name="forPipe" incrementSessionKey="i" startAt="0" max="10">
+ * 		 	<Forward name="stop" path="EXIT" />
+ * 		 	<Forward name="continue" path="echoPipe"/>
```

Comment:
Ik vind het zelf niet helemaal logisch dat hier een `<Forward/>` element wordt gebruikt om aan te geven welke pipe geloopt moet worden. Voor mij is een `<Forward/>` element iets om aan te geven waar je naar toe moet springen als exit van de pipe?

Reply:
Dit is zoals het in het ticket staat. Hoe zou je dit anders willen doen?

Reply:
Ik heb het ticket niet eerder kritisch bekeken, ik kijk nu eigenlijk alleen maar naar hoe het nu in het PR staat en krijg daarbij het gevoel dat het misschien erg ongestructureerd kan gaan worden.

Ik had zelf verwacht dat het meer zou lijken op andere IteratorPipes, dus met een genest `<sender/>` element.

Een pipe die over items in een ZIP file itereert configureer je bijvoorbeeld zo in de XML:
```xml
			<pipe
				name="NestedZipRead"
				getInputFromSessionKey="stream"
				className="org.frankframework.compression.ZipIteratorPipe"
				>
				<sender className="org.frankframework.senders.IbisLocalSender" javaListener="ibis4test-unzipNested" >
					<param name="zipdata" sessionKey="zipdata" />
				</sender>
				<forward name="success" path="READY"/>
			</pipe>
```

Ik zou hebben verwacht dat dat idioom zou worden voortgezet.

Ik snap dat jij een spec implementeert -- maar ik had daar nog niet eerder kritisch naar gekeken en heb nu mijn vragen bij die spec.

Reply:
Nog een issue wat ik zie met hoe het nu werkt: je kan nu niet garanderen dat de loop inderdaad start op de opgegeven `startAt` waarde. Als de session key per ongeluk door een andere pipe al is gezet, dan zal de loop op die waarde beginnen.

Doordat per design de flow handmatig moet terug springen naar de ForPipe, kan niet worden afgedwongen dat de loop altijd begint met zijn `startAt`.

Met een design dat een genestte sender gebruikt kan dat wel.